[Docs] Document expo doctor as a tool (#30786)
expo/expo: docs/pages/develop/tools.mdx

You can use `eas --help` in your terminal window to learn more about the available commands. For a complete reference, see [`eas-cli` npm page](https://www.npmjs.com/package/eas-cli).

## Expo Doctor

Expo Doctor is a command line tool used to diagnose issues in your Expo project. To use it, run the following command in your project's root directory:

<Terminal cmd={['$ npx expo-doctor']} />

> This command performs checks and analyzes your project's codebase for common issues, dependency compatibility, configuration files, and the overall health of the project. Once the check is complete, Expo Doctor outputs the results.

Reviewer: ```suggestion
This command performs checks and analyzes your project's codebase for common issues in [app config](/workflow/configuration/) and **package.json** files, dependency compatibility, configuration files, and the overall health of the project. Once the check is complete, Expo Doctor outputs the results.
```

After re-thinking it, I think let's explain what those common issues related to. The CLI output you shared checks these two files.

Apologies for overlooking this in my previous suggestion.

Author: Yes, I like it better. Thank you

You can use `eas --help` in your terminal window to learn more about the available commands. For a complete reference, see [`eas-cli` npm page](https://www.npmjs.com/package/eas-cli).

## Expo Doctor CLI

Expo Doctor is a command line tool used to diagnose issues in your Expo project.

To use Expo Doctor, run the following command in your project's root folder:

<Terminal cmd={['$ npx expo-doctor']} />

> This command will perform checks and analyses on your codebase, looking for common issues, dependency compatibility, configuration files, and the overall health of your project. Once the check is complete, Expo Doctor will output the results.

Reviewer: ```suggestion
This command performs checks and analyzes your project's codebase for common issues, dependency compatibility, configuration files, and the overall health of the project. Once the check is complete, Expo Doctor outputs the results.
```

- In the first sentence, I'm not sure if you meant to break it just before "looking for common issues". I think those things described here are part of the checks?
- It might not be clear to users on what can they expect when we say "common issues". Can we provide an example or two here?
- As per our [writing style guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#use-present-tense) let's prefer to use present tense in here.

Author: I wanted to mentioned like an overview of what doctor does. Heres a screenshot: 

Should we create a list of them? 

![image](https://github.com/user-attachments/assets/cd59c97a-df3a-4e21-a7e3-889d43d01e12)

Reviewer: Thanks for the clarification, it makes sense to me. I've edited my suggestion above. I think the overview sounds good to me. I don't think we should refer to the complete checklist here since this page provides an introduction to all tools with the persona in mind of a new user (maybe we can create a reference page in the future, if required, that can list all the checks).

> If Expo Doctor finds an issue, it will provide a description of the problem along with advice on how to fix it or where to find help.

Reviewer: ```suggestion
If Expo Doctor finds an issue, it provides a description of the problem along with advice on how to fix it or where to find help.
```

As per our [writing style guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#use-present-tense) let's prefer to use present tense here.
